provisioning/miniprov/hare_mp/cdf.py

```diff
@@ -141,6 +141,26 @@ def _add_pool(self, pool: PoolHandle, out_list: List[PoolDesc]) -> None:
         storage_set_name = conf.get(f'cluster>{cid}>storage_set[{i}]>name')
         pool_name = f'{storage_set_name}__{pool_type}'
 
+        enc_failure_allowed = 0
+        ctrl_failure_allowed = layout.parity
+
+        machine_id = conf.get_machine_id()
+        node_count = len(conf.get_storage_set_nodes())
+        cvg_per_node = int(conf.get(
+            f'server_node>{machine_id}>storage>cvg_count'))
+
+        cvg_per_storage_set = cvg_per_node * node_count
+        ratio = floor(cvg_per_storage_set / (
+                      layout.data + layout.parity + layout.spare))
+
+        if ratio == 0:
```

Comment:
How the formula based approach is different than what mentioned in https://jts.seagate.com/browse/EOS-19413,
```
for each pool P in pools
    if (P.disks_nr >= P.N + P.K + P.S):
        P.allowed_failures[disk] = P.K
    if (P.ctrls_nr >= P.N + P.K + P.S and
        P.ctrl.disks_nr):
        P.allowed_failures[ctrls] = P.K
        P.allowed_failures[disks] = 0
    if (P.encls_nr >= P.N + P.K + P.S):
        P.allowed_failures[encls] = P.K
        P.allowed_failures[ctrls] = 0
        P.allowed_failures[disks] = 0
```

Reply:
Also, allowed disks failure = K does not work in case of node failure if number of disks per node are > K, cc @madhavemuri.

Reply:
@mssawant : If node failure is there, failure domain code doesn't check for disk or CVG's failure below it in the configuration tree.

Reply:
@madhavemuri, yeah, looks like. I see some problems with degraded IO in case of node failure with [0, 0, 1, 2, 2]. @SwapnilGaonkar7, can you try degraded IO with the generate allowed_failures vector?
After bootstrapping the cluster, start m0client and write a file, e.g.
m0client parameters, (profile `-p` and process fid `-P`) can be  found in `hctl status`.
```
m0client -l '10.230.250.51@tcp:12345:4:1' -H '10.230.250.51@tcp:12345:1:1' -p '<0x7000000000000001:0x16e>' -P '<0x7200000000000001:0x27>'
m0client >>write 1048680 /tmp/srcfile 4096 200 50
```
Once written successfully, fail the node (reboot or crash by writing to `/proc/sysrq-trigger`, e.g. echo b > /proc/sysrq-trigger).
once node status is updated in `hctl status`, try to read the written file,
```
m0client >>read 1048680 /tmp/read_1048680  4096 200 50
```
if read is successful then degraded path is working fine. For me the read is failing.